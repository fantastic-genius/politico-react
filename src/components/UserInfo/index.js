import React from 'react';
import { connect } from 'react-redux';
import userImg from '../../assets/images/user.jpg'

const UserInfo = ({user: { firstname, lastname, passportUrl}}) => {
  return (
    <div className="user">
      <div className="photo">
        <img id="user-img" src={passportUrl || userImg} alt="User" />
      </div>
      <div className="username">
        <p id="user-name">{`${firstname} ${lastname}`}</p>
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  user: state.auth.user
})

export default connect(mapStateToProps, {})(UserInfo);
