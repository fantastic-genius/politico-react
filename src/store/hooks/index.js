import { useEffect } from 'react';
import  { setUser } from '@actions/Auth';

const useSetUser = ({ dispatch }) => {
  useEffect(() => {
    dispatch(setUser());
  }, []);
};

export default useSetUser;