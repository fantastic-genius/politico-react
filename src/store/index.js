import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { createLogger as logger } from 'redux-logger';
import promise from 'redux-promise-middleware';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import reducer from './reducers';

const middleware = applyMiddleware(promise, thunk, logger());
const store = createStore(reducer, composeWithDevTools(middleware));

export default store;
