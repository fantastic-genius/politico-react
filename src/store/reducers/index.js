import { combineReducers } from 'redux';
import auth from '@reducers/Auth';
import vote from '@reducers/Vote';
import office from '@reducers/Office';

const reducers = combineReducers({
    auth,
    vote,
    office,
});

export default reducers;
