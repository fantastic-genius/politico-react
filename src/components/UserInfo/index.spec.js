import React from 'react';
import { shallow } from 'enzyme';
import store from '@src/store';
import UserInfo from './index';

describe('<UserInfo />', () => {
  describe('Rendering', () => {
    const user = {
      firstname: 'john',
      lastname: 'mark',
      passportUrl: 'https://w.pe./img',
    }
    it('should render UserInfo correctly in debug mode', () => {
        const component = shallow(<UserInfo store={store} user={user} />);
        expect(component).toMatchSnapshot();
    });
  })

})

