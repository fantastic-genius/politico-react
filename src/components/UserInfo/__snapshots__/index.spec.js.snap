// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> Rendering should render UserInfo correctly in debug mode 1`] = `
<ContextProvider
  value={null}
>
  <UserInfo
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    user={Object {}}
  />
</ContextProvider>
`;
